### PR TITLE
Altimetric Data Block

### PR DESCRIPTION
**Pull request by charanjeetcs**

### CODE REVIEW

File: web/modules/highwire/journal_article_detail/src/Plugin/Block/AltmetricsData.php [hunk 1]
```diff
@@ -0,0 +1,195 @@
...
+   */
+  public function blockSubmit($form, FormStateInterface $form_state) {
+    parent::blockSubmit($form, $form_state);
+    $values = $form_state->getValues();
+    $data = ['label', 'label_display', 'description', 'month', 'issns', 'number_per_page', 'api_url', 'show_pager', 'more_link', 'more_link_label', 'more_link_url', 'msg_no_data'];
+    foreach ($data as $option) {
+      $this->configuration[$option] = $values[$option];
+    }
+  }
+}
```
Comment (RahulSatija49) [inline web/modules/highwire/journal_article_detail/src/Plugin/Block/AltmetricsData.php:195]: Need a blank line end of file

File: web/modules/highwire/journal_article_detail/templates/altmetrics-data.html.twig [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+{% else %}
+    {{ altmetrics_data.msg_no_data }}
+{% endif %}
+
+{% if altmetrice_setting.more_link == 1 %}
+    <a href="{{ altmetrice_setting.more_link_url }}">{{ altmetrice_setting.more_link_label }}</a>
+{% endif %}
+{% if altmetrice_setting.show_pager == 1 %}
+    {{ pager }}
+{% endif %}
```
Comment (RahulSatija49) [inline web/modules/highwire/journal_article_detail/templates/altmetrics-data.html.twig:34]: Need a blank line end of file

File: web/modules/highwire/journal_article_detail/src/Plugin/Block/AltmetricsData.php [hunk 3]
```diff
@@ -0,0 +1,195 @@
...
+      $build = [
+        '#theme' => 'altmetrics_data',
+        '#altmetrics_data' => $results,
+        '#altmetrice_setting' => $config,
+      ];
+    }
+    
+    $build['#attached']['library'][] = 'journal_article_detail/altmetrics';
+
+    //dd($altmetrics_data);
```
Comment (RahulSatija49) [inline web/modules/highwire/journal_article_detail/src/Plugin/Block/AltmetricsData.php:86]: remove unwanted data and space.

File: web/modules/highwire/journal_article_detail/journal_article_detail.module [hunk 4]
```diff
@@ -136,6 +136,9 @@ function journal_article_detail_theme() {
 		],
     'hwjma_usage_stats_data' => [
       'variables' => ['filter_tabs' => NULL, 'usage_stats_data' => NULL]
+    ],
+    'altmetrics_data' => [
+      'variables' => ['altmetrics_data' => NULL, 'altmetrice_setting' => NULL, 'pager' => NULL]
     ]
 	];
 }
```
Comment (RahulSatija49) [inline web/modules/highwire/journal_article_detail/journal_article_detail.module:144]: Need a blank line end of file

**Review by RahulSatija49 — COMMENTED** [review]
Please fix below points

File: web/modules/highwire/journal_article_detail/src/Plugin/Block/AltmetricsData.php (hunk 1 above)
Comment (charanjeetcs) [reply web/modules/highwire/journal_article_detail/src/Plugin/Block/AltmetricsData.php]: Resolved

File: web/modules/highwire/journal_article_detail/templates/altmetrics-data.html.twig (hunk 2 above)
Comment (charanjeetcs) [reply web/modules/highwire/journal_article_detail/templates/altmetrics-data.html.twig]: Resolved

File: web/modules/highwire/journal_article_detail/src/Plugin/Block/AltmetricsData.php (hunk 3 above)
Comment (charanjeetcs) [reply web/modules/highwire/journal_article_detail/src/Plugin/Block/AltmetricsData.php]: Resolved

File: web/modules/highwire/journal_article_detail/journal_article_detail.module (hunk 4 above)
Comment (charanjeetcs) [reply web/modules/highwire/journal_article_detail/journal_article_detail.module]: Resolved

**Pull request merged**